fendermint/vm/interpreter/src/fvm/broadcast.rs

```diff
@@ -0,0 +1,143 @@
+// Copyright 2022-2023 Protocol Labs
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use anyhow::{anyhow, bail, Context};
+use ethers::types as et;
+use fendermint_rpc::message::GasParams;
+use fendermint_rpc::query::QueryClient;
+use fendermint_rpc::tx::{CallClient, TxClient, TxCommit};
+use fendermint_rpc::{client::FendermintClient, message::MessageFactory};
+use fendermint_vm_message::query::FvmQueryHeight;
+use fvm_shared::{address::Address, chainid::ChainID, econ::TokenAmount, BLOCK_GAS_LIMIT};
+use libsecp256k1::{PublicKey, SecretKey};
+use num_traits::Zero;
+use tendermint_rpc::Client;
+
+/// Broadcast transactions to Tendermint.
+///
+/// This is typically something only active validators would want to do
+/// from within Fendermint as part of the block lifecycle, for example
+/// to submit their signatures to the ledger.
+///
+/// The broadcaster encapsulates the tactics for figuring out the nonce,
+/// the gas limit, potential retries, etc.
+#[derive(Clone)]
+pub struct Broadcaster<C> {
+    client: FendermintClient<C>,
+    secret_key: SecretKey,
```

Comment:
Do we zeroize these secret bytes?

Reply:
Can you elaborate? I have an idea of what you mean, that if a program crashes it should not be recoverable from a memory dump, however I don't see when I could do this given that it's needed for each submission. What is the recommended way of using secret keys? Should it be read from disk every time? Should it be protected by a password? How should that password be protected?

Reply:
You are right. The best description of the problem can be found here - https://github.com/veorq/cryptocoding#clean-memory-of-secret-data

In Rust, zeroize - https://crates.io/crates/zeroize - is a de facto standard for clearing memory.

And this is an example of usage of `zeroize` - https://github.com/matrix-org/vodozemac/pull/43/files

Reply:
Thanks, I created https://github.com/consensus-shipyard/fendermint/issues/279 to deal with this.